src/m3ninx/index/segment/fst/segment.go

```diff
@@ -612,6 +617,37 @@ func (r *fsSegment) docsNotClosedMaybeFinalizedWithRLock(
 	return index.NewIDDocIterator(retriever, pl.Iterator()), nil
 }
 
+func (r *fsSegment) encodedDocNotClosedMaybeFinalizedWithRLock(id postings.ID) (doc.EncodedDocument, error) {
+	// NB(r): Not closed, but could be finalized (i.e. closed segment reader)
+	// calling match field after this segment is finalized.
+	if r.finalized {
+		return doc.EncodedDocument{}, errReaderFinalized
+	}
+
+	// TODO(nate): support for thirdPartyReader? seems to only to be used for compaction which
+	// is unnecessary for encoded docs use case of read queries.
```

Comment:
The non-encoded version of this method supports using a `docsThirdPartyReader`: https://github.com/m3db/m3/blob/master/src/m3ninx/index/segment/fst/segment.go#L590

That doesn't seem applicable here, but wanted to ask this question